src/repr/src/row.rs

```diff
@@ -259,6 +382,109 @@ mod columnation {
     }
 }
 
+/// A contiguous slice of bytes that are row data.
+///
+/// A [`RowRef`] is to [`Row`] as [`prim@str`] is to [`String`].
+#[derive(PartialEq, Eq)]
+#[repr(transparent)]
+pub struct RowRef([u8]);
+
+impl RowRef {
+    /// Create a [`RowRef`] from a slice of data.
+    ///
+    /// We do not check that the provided slice is valid [`Row`] data, will panic on read
+    /// if the data is invalid.
+    pub fn from_slice(row: &[u8]) -> &RowRef {
```

Comment:
@teskje and I went back and forth on this. Originally it was marked as unsafe but we opted to remove that since trying to read invalid Row data will panic if it's malformed. Also you can do this byte -> RowRef conversion by creating a ProtoRowCollection which does not require any unsafe

More than happy to re-add the unsafe if we think that's the best call

Reply:
Late add: I think it's somewhat important that this remains not `unsafe`. If we made `from_slice` unsafe and documented that callers must provide a correct encoding, then future developers could be tempted to make the wrong assumption that a correct encoding is an invariant for the `RowRef` data and write code that is UB if it is not. But as Parker mentioned, proto decoding will always let us get invalidly encoded bytes into a `RowRef`, and we can't declare that `unsafe`, so this invariant can never be assumed.

Of course we could (and maybe should) make it explicit in the docstring of `RowRef` that the wrapped data might be invalid and all using code needs to be able to handle that. But even then there is nothing unsafe about calling `from_slice` with invalid data, so no reason to make it an `unsafe` method.